### PR TITLE
fence_openstack: Migrate from keystoneclient to keystoneauth1

### PR DESCRIPTION
**Pull request by racciari**
keystoneclient is deprecated in favor of keystoneauth1.

### CODE REVIEW

**Comment by oalbrigt** [comment]
Can you add fallback to keystoneclient, so it still works for systems with keystoneclient?

**Comment by racciari** [comment]
Here you go, but I've not tested the legacy import with keystoneclient.

File: agents/openstack/fence_openstack.py [hunk 1]
```diff
@@ -11,12 +11,15 @@
...
 try:
-        from keystoneclient.v3 import client as ksclient
+        from novaclient import client as novaclient
+        from keystoneauth1 import session as ksc_session
+        from keystoneauth1 import loading
+        legacy_import = False
+except ImportError:
         from novaclient import client as novaclient
         from keystoneclient import session as ksc_session
         from keystoneclient.auth.identity import v3
```
Comment (oalbrigt) [inline agents/openstack/fence_openstack.py]: I think you meant to put `legacy_import=True` here.

Comment (racciari) [reply agents/openstack/fence_openstack.py]: The `legacy_import=True` is in the `ImportError` section below

**Comment by oalbrigt** [comment]
ok to test

File: agents/openstack/fence_openstack.py (hunk 1 above)
Comment (oalbrigt) [reply agents/openstack/fence_openstack.py]: It needs some extra nesting, so I'll take care of that after merging.

**Comment by oalbrigt** [comment]
Created https://github.com/ClusterLabs/fence-agents/pull/248 to add the except: pass back for metadata generation, and also added keystonauth1 to our library detection.

**Pull request merged**